### PR TITLE
fix crate dragging with traders

### LINKED ISSUE #2061: Trader click-drag doesn't work.
Submitter: Brian Renadette
Using the click-drag on a crate to sell multiple items to a physical trader doesn't work. I tried selling a crate of omega weed to the drug-dealing robot in the Space Diner, but I'm told there's nothing in the crate that interests it.

Reported on: Goonstation 2020-09-04 08:58:38

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
Apparently this has been broken for a while?
There is a pinned issue for it, to be honest, I never see those because I never look to the pinned issue area.
fixes #2061
@pali6 writes that it should be it's own proc, honestly, I don't see the need for that, it seems like a simple typecheck to me but what do I know.
Anyway, even if the way I fixed it is bad, it's probably better than leaving it broken for another 5 months.^^

## Why's this needed? 
It's a pretty cool feature!

## Changelog

```changelog
(u)zjdtmkhzt
(+)Dragging crates onto traders to sell their contents should now work again for all types of items.
```

### CODE REVIEW

**Comment by zjdtmkhzt** [comment]
Is the issue something like white weed vs normal weed?
In that case we could either just take care with the order in which commodities are stored, or we first check for exact matches and then just types.
Not sure if we already have a thing like that for regular trades? I didn't see one.

**Comment by zjdtmkhzt** [comment]
Yeah, lol, don't merge this yet, right now you get the money for all the types your item applies to.^^

**Comment by zjdtmkhzt** [comment]
ok, done, moved it to its own proc!

File: code/modules/economy/trader.dm [hunk 1]
```diff
@@ -515,16 +518,10 @@
 				playsound(src.loc, "rustle", 60, 1)
 				var/cratevalue = null
 				for (var/obj/M in O.contents)
```
Comment (ZeWaka) [inline code/modules/economy/trader.dm:520]: M? For an OBJ????

File: code/modules/economy/trader.dm [hunk 2]
```diff
@@ -1,7 +1,15 @@
+/proc/most_applicable_trade(var/list/goods_buy, var/obj/item/sell_item)
+	var/list/goods_buy_types = new /list(0)
+	for(var/datum/commodity/N in goods_buy)
```
Comment (ZeWaka) [inline code/modules/economy/trader.dm:3]: specify the type of list in the proc argument and make this an `as()`

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Review by pali6 — APPROVED** [review]

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**